docs/cmdline-opts/get.md

```diff
@@ -22,7 +22,11 @@ Example:
 When used, this option makes all data specified with --data, --data-binary
 or --data-urlencode to be used in an HTTP GET request instead of the POST
 request that otherwise would be used. The data is appended to the URL
-with a '?' separator.
+with a '?' as a separator. Or with '&' if more data is provided (check
+--data for details).
```

Comment:
This is not true.`?` is a query separator. The ampersand (`&`) is inserted between multiple data strings provided on the command line. Different things.

Reply:
I wanted to point that curl will use appropriate separator depending on number of data:
- `?` if only one data is to be appended to URL
- `?` and/or `&` if more data is to be appended or there is already some data in URL. `&` can be considered as separator between two data fields in query string.

I writing `?` **and/or** `&` because of following cases:
1. User already provided data in URL, so curl only used `&`
```shell
$ curl --silent --get --data key=value 'https://httpbin.org/get?foo=bar' -v
[...]
> GET /get?foo=bar&key=value HTTP/1.1
[...]
```
2. User provided all data as command line options, so curl used both `?` (to separate query string from main part of URL) and `&` (to separate data fields):
```shell
$ curl --silent --get --data key=value --data foo=bar 'https://httpbin.org/get' -v
[...]
> GET /get?key=value&foo=bar HTTP/1.1
[...]
```

My colleague was afraid that `?` will be used always as per current statement:
>  The data is appended to the URL with a '?' separator.

Maybe it's not worth mentioning those separators at all? curl will simply do the proper thing 😉

Reply:
The ampersand (`&`) is used as a separator between multiple parts of query pieces. That is described in section for `--data`.

When all of that data is added to the URL, it is put as a *query* component. Query components are always provided in the URL with a `?` separator. This latter part is what `--get` does, and that is why it is added in the documentation for --get.

Reply:
Hi @bagder!  
I pushed new commit with simplified description of `--get` / `-G` option.

Why I wanted update in original description? With colleague we were preparing a command to send some data to API endpoint. I prepared it working on 8.4.0 and it worked. He was using some older version of curl (prior to 7.86.0) and for him same command was not working. Command was something like this:
```shell
$ cat file.txt | curl --get --data-urlencode key@- 'http://example.com?foo=bar&data=leading/to/corner/case'
```
He was debugging and found that for him final URL is:
```text
GET /?foo=bar&data=leading/to/corner/case?key=world%0A HTTP/1.1
     ^                                   ^
```
He [checked the documentation](https://curl.se/docs/manpage.html#-G) and found:
> The data is appended to the URL with a '?' separator.

So he assumed that whole idea is wrong, as curl will always use `?`.

Of course I was surprised that it does not work for him, so I started looking if it was corrected and when.  
Knowing it, I wanted to update curl manual to nuance a bit what separator will be used. And to document that there was a change in 7.86.0 which fixed one problematic corner case.

I hope knowing this story, you will better understand my motivation 😄 

PS. As it can happen that in our env there will be more people with curl older that 7.86.0 we finally switched to:
```shell
$ cat file.txt | curl --get --data-urlencode key@- --data 'foo=bar' --data 'data=leading/to/corner/case' 'http://example.com'
```